http-client/src/main/java/io/micronaut/http/client/exceptions/HttpClientResponseException.java

```diff
@@ -56,21 +55,20 @@ public HttpClientResponseException(String message, Throwable cause, HttpResponse
      * @param response The Http response
      * @param errorDecoder The error decoder
      */
-    public HttpClientResponseException(String message, Throwable cause, HttpResponse<?> response, HttpClientErrorDecoder errorDecoder) {
+    public HttpClientResponseException(String message, HttpRequest<?> request, Throwable cause, HttpResponse<?> response, HttpClientErrorDecoder errorDecoder) {
         super(message, cause);
         this.errorDecoder = errorDecoder;
         this.response = response;
+        this.request = request;
         initResponse(response);
     }
 
     @Override
     public String getMessage() {
         Optional<Argument<?>> errorType = Optional.ofNullable(getErrorType(response));
-        if (errorType.isPresent()) {
-            return getResponse().getBody(errorType.get()).flatMap(errorDecoder::getMessage).orElse(super.getMessage());
-        } else {
-            return super.getMessage();
-        }
+        return "Request ["+ this.request.getUri().getPath() +"] failed with error: "+ errorType.map(errType ->
+             getResponse().getBody(errorType.get()).flatMap(errorDecoder::getMessage).orElse(super.getMessage())
+        ).orElse(super.getMessage());
```

Comment:
This change should be reverted. This removes control from the error decoder for the full message

Reply:
Since we want properties of requests to be returned in all HttpClientResponseExceptions, I had done it here. 
Should I move it to error decoder instead? In that case, those request properties won't be returned for responses without a type(HttpClientException/HttpException).

Reply:
Just because you have added access to the request does not mean we should be changing the exception message

Reply:
Since this is a PR for https://github.com/micronaut-projects/micronaut-core/issues/2240, I had modified the message to include the requestURI.

Reply:
@ashwini-desai We can't accept this PR without reverting this change